src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpThurloeDAO.scala

```diff
@@ -84,15 +84,14 @@ class HttpThurloeDAO ( implicit val system: ActorSystem, implicit val executionC
     * @param callerToken the OAuth token of the person making the API call
     * @return the trial status for the specified user, or None if trial status could not be determined.
     */
-  override def getTrialStatus(forUserId: String, callerToken: WithAccessToken): Future[Option[UserTrialStatus]] = {
+  override def getTrialStatus(forUserId: String, callerToken: WithAccessToken): Future[UserTrialStatus] = {
     wrapExceptions {
       userAuthedRequest(Get(UserApiService.remoteGetAllURL.format(forUserId)), false, true)(callerToken) map { response =>
         response.status match {
           case StatusCodes.OK =>
             val status = UserTrialStatus(unmarshal[ProfileWrapper].apply(response))
             assert(forUserId == status.userId, "status id does not match!")
-            Some(status)
-          case StatusCodes.NotFound => None
+            status
```

Comment:
The `userAuthedRequest` will only return an error if the user has no KVPs in Thurloe at all ... which we *should* consider an error, since we expect all free-trial users to already be registered and have filled out their profile.

As long as Thurloe returns at least one KVP, the `response.status` will be `OK`.

Reply:
actually thurloe will return an empty key value list even if the user doesn't exist in thurloe - for example:
{
  "userId": "asdf",
  "keyValuePairs": []
}

Reply:
Seeing the same Thurloe response as @ahaessly described. Doesn't this conflict with your reasoning @davidangb? 

Added a [test](https://github.com/broadinstitute/firecloud-orchestration/pull/573/commits/74164d698b5688bd79fae90a02fea9c775fab70a) in PR#579 for this scenario.